bin/faucet/src/faucet/mod.rs

```diff
@@ -281,177 +177,51 @@ impl Faucet {
-        account.apply_delta(delta)?;
-        self.data_store.update_faucet_state(account);
-
-        Ok(())
-    }
-
-    /// Attempt to recover from errors.
-    ///
-    /// Notably this includes rolling back local state if a desync occurs.
-    ///
-    /// Returns an error if recovery was not possible, which should be considered fatal.
-    fn error_recovery(&mut self, err: MintError) -> anyhow::Result<()> {
-        match err {
-            // A state mismatch means we desync'd from the actual chain state, and should resync.
-            //
-            // This can occur if the node restarts (dropping inflight txs), or if inflight txs got
-            // dropped.
-            MintError::Submission(RpcError::Transport(err))
-                if err.code() == tonic::Code::InvalidArgument
-                    && err.message().contains("incorrect initial state commitment") =>
-            {
-                self.handle_desync(err.message()).with_context(|| {
-                    format!("failed to recover from desync error: {}", err.message())
-                })
-            },
-            // TODO: Look into which other errors should be recoverable.
-            //       e.g. Connection error being lost to RPC client is probably not fatal.
-            others => Err(others).context("failed to handle error"),
+            let block_height = self.client.get_sync_height().await?;
+            updater.send_notes(block_height, &notes, tx_id).await;
         }
-    }
 
-    /// Attempts to rollback back local state to match that indicated by the node.
-    ///
-    /// This relies on parsing the stringified error
-    /// `VerifyTxError::IncorrectInitialAccountCommitment`.
-    ///
-    /// Returns an error if the rollback was unsuccessful. This should be treated as fatal.
-    fn handle_desync(&mut self, err: &str) -> anyhow::Result<()> {
-        let onchain_state = parse_desync_error(err).context("failed to parse desync message")?;
-
-        // Find the matching local state, unless we've dropped it already.
-        let rollback = self
-            .prior_state
-            .iter()
-            .position(|state| state.commitment() == onchain_state)
-            .context("no matching local state to rollback to")?;
-
-        // Rollback the local state.
-        // SAFETY: The index must exist since the element was just found.
-        self.data_store.update_faucet_state(self.prior_state[rollback].clone());
-        self.prior_state.drain(rollback..);
-
-        tracing::warn!(rollback.count = rollback, "desync detected and handled");
+        tracing::info!("Request stream closed, shutting down minter");
 
         Ok(())
     }
 
-    /// Fully handles a batch of requests _without_ changing local state.
-    ///
-    /// Caller should update the local state based on the returned result.
+    /// Fully handles a batch of requests to create and submit a transaction, and updates the local
+    /// db.
```

Comment:
I would explain here what this involves (for example, getting multiple requests and submitting notes for them, updating clients in the meantime, updating the local Miden client, etc.)